src/Views/Admin/NewDog.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useState } from 'react';
+import DogForm from '../../Components/DogForm/DogForm';
+import { addANewDog } from '../../services/dogs';
+import './NewDog.css';
+
+export default function NewDog() {
+  const [dog, setDog] = useState({ name: '', age: '', breed: '', bio: '', image: '' });
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState('');
+  const history = useHistory();
+
+  const handleSubmit = async () => {
+    try {
+      await addANewDog(dog);
+      setSuccess(true);
+      setTimeout(() => {
+        history.push('/');
+      }, 2000);
```

Comment:
Be sure to use clearTimout when calling setTimeout to avoiding memory leaks when a component unmounts.